### PR TITLE
Added support to export cmake targets

### PR DESCRIPTION
**Pull request by Bergmann89**
I use the export functionality of cmake in my projects. If I use the json-schema-validator in my projects, cmake prints the following error.

`CMake Error: install(EXPORT "my-project" ...) includes target "my-project-static" which requires target "json-schema-validator" that is not in the export set.`

To fix this, the json-schema-validator must export it's targets too. The pull request fixes the error I have with my projects, but I think there must be added something more to fully support the cmake exports.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -117,3 +119,7 @@ if (BUILD_TESTS)
     enable_testing()
     add_subdirectory(test)
 endif()
+
+install ( EXPORT json-schema-validator
```
Comment (pboettch) [inline CMakeLists.txt]:
```suggestion
install(EXPORT json-schema-validator
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -117,3 +119,7 @@ if (BUILD_TESTS)
...
+          NAMESPACE json-schema-validator::
```
Comment (pboettch) [inline CMakeLists.txt]:
```suggestion
        NAMESPACE json-schema-validator::
```

**Comment by schaumb** [comment]
I tried to use this patch with [Hunter](https://github.com/ruslo/hunter), but it's says:

`CMake Error: install(EXPORT "json-schema-validator" ...) includes target "json-schema-validator" which requires target "json-hpp" that is not in the export set.`

**Comment by pboettch** [comment]
Please feel free to check with #73 which is an updated version of #71 and a more modern version of yours.

**Pull request closed without merging**